src/component/video-card/VideoCard.jsx

```diff
@@ -4,82 +4,123 @@ import WatchLaterOutlinedIcon from '@mui/icons-material/WatchLaterOutlined';
 import PlayCircleOutlineOutlinedIcon from '@mui/icons-material/PlayCircleOutlineOutlined';
 import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
 
-import { useNavigate } from 'react-router-dom';
+import { useNavigate, useLocation } from 'react-router-dom';
 import { useState } from 'react';
 
-import { addItemToWatchLater, removeItemFromWatchLater } from 'service';
+import { addItemToWatchLater, removeItemFromWatchLater, addVideoToHistory, removeVideoFromHistory } from 'service';
 import { useData, useAuth } from 'context';
 import { useToast } from 'custom-hook/useToast';
 
-const VideoCard = ({video}) => {
+const VideoCard = ({ video }) => {
   const navigate = useNavigate();
-  const { videoDispatch, videoState:{watchlater}} = useData();
-  const { auth:{isAuth} } = useAuth();
+  const location = useLocation();
+  const { videoDispatch, videoState: { watchlater, history } } = useData();
+  const { auth: { isAuth } } = useAuth();
   const { showToast } = useToast();
   const [optionModal, setOptionalModal] = useState(false);
-  const optionHandler = e =>{
+  const optionHandler = e => {
     e.stopPropagation();
     setOptionalModal(prevModalState => !prevModalState);
   }
   const isVideoInWatchLater = () => watchlater.find(item => item._id === video._id) ? true : false;
-  const watchLaterIcon = isVideoInWatchLater() ? <DeleteOutlineOutlinedIcon style={{fontSize: "1rem"}}/> : <WatchLaterOutlinedIcon style={{fontSize: "1rem"}}/>;
+  const isVideoInHistory = () => history.find(item => item._id === video._id) ? true : false;
```

Comment:
these two functions have same code, only difference is the array in which you need to check the video, so this can be extracted out as a utility function and pass in the video and array as parameter accordingly

Reply:
Yes makes sense. I'll refactor here as well.